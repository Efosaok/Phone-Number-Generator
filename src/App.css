.App {
  text-align: center;
  font-family: verdana;
}

.App-link {
  color: #61dafb;
}

.number-list {
  padding-top: 10px;
  justify-content: center;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
}

.number {
  margin-bottom: 5px;
  margin-right: 10px;
}

.number-list-action-section {
  display: flex;
  justify-content: space-between;
  flex-direction: row;
  flex-wrap: wrap;
  padding-top: 20px;
  padding-left: 600px;
  padding-right: 600px;
}

.sort-action {
  display: flex;
  flex-direction: row;
}

.sort-label {
  font-size: 15px;
}

.max-btn-section, .min-btn-section {
  margin-top: -3px;
}
