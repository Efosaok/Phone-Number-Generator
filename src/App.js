import React, { Component } from 'react';
import GenerateNumbers from './GenerateNumbers';

import './App.css';

class App extends Component {
  render() {
    return (
      <div className="App">
        <div className="header">
          <h3>Generate Phone Numbers</h3>
        </div>
        <GenerateNumbers />
      </div>
    );
  }
}

export default App;
