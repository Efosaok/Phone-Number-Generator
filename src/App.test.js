import React from 'react';
import App from './App';

import { render, fireEvent,getAllByTestId, getByTestId, cleanup } from 'react-testing-library';

describe('App test suite', () => {
  const wrapper = render(<App />)
  const { getByText, getByTestId } = wrapper;

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('generates number when the generate number button is clicked', () => {
    const generateNumberButton = getByText('Generate Numbers');
    const numbersListSection = getByTestId('number-list-section');
    fireEvent.click(generateNumberButton);

    expect(numbersListSection).toBeTruthy();
  });

  it('shows the maximum number when the max button is clicked', () => {
    const generateNumberButton = getByText('Generate Numbers');
    fireEvent.click(generateNumberButton);
    const maxButton = getByText('max');
    const number = getByTestId('number');
    fireEvent.click(maxButton);

    expect(number).toBeTruthy();
  });

  it('shows the minimum number when the min button is clicked', () => {
    const generateNumberButton = getByText('Generate Numbers');
    fireEvent.click(generateNumberButton);
    const minButton = getByText('min');
    const number = getByTestId('number');
    fireEvent.click(minButton);

    expect(number).toBeTruthy();
  });

  it('sorts numbers when the select value is changed to ascending', () => {
    const generateNumberButton = getByText('Generate Numbers');
    fireEvent.click(generateNumberButton);
    const numbersListSection = getByTestId('number-list-section');

    const selectDropdown = getByTestId('select');
    fireEvent.change(selectDropdown, { target: { value: 'ascending' }})

    expect(numbersListSection).toBeTruthy();
  });

  it('sorts numbers when the select value is changed to descending', () => {
    const generateNumberButton = getByText('Generate Numbers');
    fireEvent.click(generateNumberButton);
    const numbersListSection = getByTestId('number-list-section');

    const selectDropdown = getByTestId('select');
    fireEvent.change(selectDropdown, { target: { value: 'descending' }})

    expect(numbersListSection).toBeTruthy();
  });
});
