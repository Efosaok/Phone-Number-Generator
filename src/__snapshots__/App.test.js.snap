// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test suite renders properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <div
          class="header"
        >
          <h3>
            Generate Phone Numbers
          </h3>
        </div>
        <div
          class="generate-number-section"
        >
          <div
            class="gen-input-section"
          >
            <input
              class="gen-input"
              placeholder="how many numbers, default is 25, max is 5000"
              type="number"
            />
          </div>
          <div
            class="action-section"
          >
            <button
              class="input gen-numbers"
            >
              Generate Numbers
            </button>
            <a
              class="input download-numbers"
              download="numbers.txt"
              href="data:text/plain;charset=utf-8,"
            >
              <button>
                download numbers
              </button>
            </a>
          </div>
          <div
            class="number-list"
            data-testid="number-list-section"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <div
        class="header"
      >
        <h3>
          Generate Phone Numbers
        </h3>
      </div>
      <div
        class="generate-number-section"
      >
        <div
          class="gen-input-section"
        >
          <input
            class="gen-input"
            placeholder="how many numbers, default is 25, max is 5000"
            type="number"
          />
        </div>
        <div
          class="action-section"
        >
          <button
            class="input gen-numbers"
          >
            Generate Numbers
          </button>
          <a
            class="input download-numbers"
            download="numbers.txt"
            href="data:text/plain;charset=utf-8,"
          >
            <button>
              download numbers
            </button>
          </a>
        </div>
        <div
          class="number-list"
          data-testid="number-list-section"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
